### PR TITLE
Advanced search fix/Certification fix/Patron disappearing fix

### PR DESCRIPTION
**Pull request by AboodV2**

### CODE REVIEW

File: app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml [hunk 1]
```diff
@@ -15,6 +15,8 @@
     {
         string MemberID = ViewBag.MemberId;
         <div class="col-sm-8 col-sm-offset-2 col-xs-10 col-xs-offset-1">
+            @if (Model.AccountType == UserAccountType.Patron)
```
Comment (dubeme) [inline app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml]: This also need to go into ViewCertification.cshtml

Comment (sakurano92) [inline app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml]: why are you doing this? doesnt non-patron also uses the wall and im pretty sure they have certification about it?

**Review by sakurano92 — COMMENTED** [review]
please check comment with in hide cerification tab

File: app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml (hunk 1 above)
Comment (AboodV2) [reply app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml]: dubeme told me to do this.According to him,Patrons are the only group that use the certification feature

Comment (dubeme) [reply app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml]: You guys have to stop the excuse of, "**Dubem told me to**".

We're all part of this project, hence should understand the requirements.

**Review by dubeme — CHANGES_REQUESTED** [review]

File: app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml (hunk 1 above)
Comment (quazi-irfan) [reply app/wama.web/Views/AdminConsole/UserTool/ViewAccount.cshtml]: I ditto @sakurano92

File: app/wama.web/Controllers/UserToolController.cs [hunk 2]
```diff
@@ -236,7 +236,8 @@ public IActionResult AddCertification(string memberId)
         public async Task<IActionResult> ViewCertifications(string memberId)
         {
             SetActiveConsoleTool(Constants.ADMIN_CONSOLE_USERS_PATRONS);
-            var Certifications = await _CertificationService.GetCertificationsAsync(memberId);
+            var Certifications = await _UserAccountService.GetUserAccountAsync(memberId);
```
Comment (dubeme) [inline app/wama.web/Controllers/UserToolController.cs]: Not bad

Comment (dubeme) [reply app/wama.web/Controllers/UserToolController.cs]: But, things can be improved.

Why are you setting a variable named **_Certifications_** to a **_UserAccount._**?

File: app/wama.web/Controllers/UserToolController.cs [hunk 3]
```diff
@@ -236,7 +236,8 @@ public IActionResult AddCertification(string memberId)
...
+            ViewData["Certifications"] = await _CertificationService.GetCertificationsAsync(memberId);
```
Comment (dubeme) [inline app/wama.web/Controllers/UserToolController.cs]: Why are you switching to passing the **_Certifications_** via **_ViewData_**?

Comment (dubeme) [reply app/wama.web/Controllers/UserToolController.cs]: Being that we are in the **_ViewCertifications_** action, one would expect that this will return a view that takes a collection of **_Certification_**.

But, you seem to be passing a member to the view instead. This isn't the best software engineering. 

If all you need is a set of extra values that will help in generating your view, then that's what the **_ViewBag_**, or **_ViewData_** is for. The **_model_** parameter you pass to **_View_** should be the main object of interest. In this case, a collection of **_Certification_**.

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertification.cshtml [hunk 4]
```diff
@@ -1,9 +0,0 @@
-﻿@using WAMA.Core.ViewModel;
```
Comment (dubeme) [inline app/wama.web/Views/AdminConsole/UserTool/ViewCertification.cshtml]: Why is this file removed?

File: app/wama.web/Views/Shared/ViewCertification.cshtml [hunk 5]
```diff
@@ -0,0 +1,58 @@
+﻿@using WAMA.Core.ViewModel;
```
Comment (dubeme) [inline app/wama.web/Views/Shared/ViewCertification.cshtml]: This file isn't needed.

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml [hunk 6]
```diff
@@ -3,57 +3,29 @@
 @using WAMA.Web.Controllers;
 @using WAMA.Web.Model;
 @using WAMA.Core.Extensions;
-
-@model IEnumerable<CertificationViewModel>
```
Comment (dubeme) [inline app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml]: A lot of the changes of this file has to be reverted.

Comment (dubeme) [reply app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml]: Once you pass in the data properly, there'll be no need for removing all the code that was removed.

**Review by dubeme — CHANGES_REQUESTED** [review]
Not bad.

But, somethings needs some work.

File: app/wama.web/Controllers/UserToolController.cs (hunk 3 above)
Comment (sakurano92) [reply app/wama.web/Controllers/UserToolController.cs]: I am pretty sure, i put var certification to grab the certification data of some user, why did you change to grab User account into certification??
use of view data should be only used for passing info into view (html)

put it back please

if you requires of grabbing user account service, use a new var instead of changing certifications

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml (hunk 6 above)
Comment (AboodV2) [reply app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml]: Fixed

File: app/wama.web/Views/Shared/ViewCertification.cshtml (hunk 5 above)
Comment (AboodV2) [reply app/wama.web/Views/Shared/ViewCertification.cshtml]: Fixed

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertification.cshtml (hunk 4 above)
Comment (AboodV2) [reply app/wama.web/Views/AdminConsole/UserTool/ViewCertification.cshtml]: Fixed

File: app/wama.web/Controllers/UserToolController.cs (hunk 3 above)
Comment (AboodV2) [reply app/wama.web/Controllers/UserToolController.cs]: Fixed

File: app/wama.web/Controllers/UserToolController.cs (hunk 2 above)
Comment (AboodV2) [reply app/wama.web/Controllers/UserToolController.cs]: Fixed

File: app/wama.web/Controllers/UserToolController.cs [hunk 7]
```diff
@@ -237,16 +237,19 @@ public IActionResult AddCertification(string memberId)
         {
             SetActiveConsoleTool(Constants.ADMIN_CONSOLE_USERS_PATRONS);
             var Certifications = await _CertificationService.GetCertificationsAsync(memberId);
+            ViewBag.AccountType = await _UserAccountService.GetUserAccountAsync(memberId);
```
Comment (dubeme) [inline app/wama.web/Controllers/UserToolController.cs]: Looks much better :)

Comment (dubeme) [reply app/wama.web/Controllers/UserToolController.cs]: Although, I feel the name of the property in the **_ViewBag_** could be more precise, since you're storing a **_UserAccount_** in it.

Maybe, change the name to **_ViewBag.UserAccount_**.

Or, perform all the logical checks in the action

```
var account =  await _UserAccountService.GetUserAccountAsync(memberId);
ViewBag.IsPatron = Equals(account, null) ? false : account.AccountType == UserAccountType.Patron;
```

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml [hunk 8]
```diff
@@ -9,51 +9,59 @@
...
-                    <tr>
-                        <th>@Html.LabelFor(x => x.ElementAt(0).Type)</th>
-                        <th>@Html.LabelFor(x => x.ElementAt(0).CertifiedOn)</th>
-                        <th>@Html.LabelFor(x => x.ElementAt(0).ExpiresOn)</th>
-                    </tr>
-                </thead>
-                <tbody>
-                    @foreach (var item in Model)
-                    {
+    if (object.Equals((ViewBag.AccountType).AccountType, UserAccountType.Patron))
```
Comment (dubeme) [inline app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml]: If you choose my 2nd suggested code, then you do something like this.

`if ((bool)(ViewBag.IsPatron??false))`

**Review by dubeme — CHANGES_REQUESTED** [review]
Much better

File: app/wama.web/Controllers/UserToolController.cs (hunk 7 above)
Comment (AboodV2) [reply app/wama.web/Controllers/UserToolController.cs]: addressed

File: app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml (hunk 8 above)
Comment (AboodV2) [reply app/wama.web/Views/AdminConsole/UserTool/ViewCertifications.cshtml]: addressed

**Review by dubeme — COMMENTED** [review]
Looking good. Will do another run through before accepting.

Remember to go review other peoples PR.

**Review by dubeme — APPROVED** [review]
